openfisca_core/taxscales.py

```diff
@@ -58,6 +58,13 @@ def copy(self):
         new.__dict__ = copy.deepcopy(self.__dict__)
         return new
 
+    def to_dict(self):
+        values = self.amounts if isinstance(self, AmountTaxScale) else self.rates
```

Comment:
```suggestion
        #  TODO: Harmonise 'rates' and 'amounts' protocols, in order to avoid this kind of check.
        values = self.amounts if isinstance(self, AmountTaxScale) else self.rates
```